src/eval-service.js

```diff
@@ -9,11 +7,13 @@ class EvalService {
             $event: options.$event
         };
 
-        if(options.attrs) {
+        if (options.attrs) {
             locals.attrs = options.attrs;
         }
 
-        scope.$eval(options.fn, locals);
+        scope.$apply(() => {
+            scope.$eval(options.fn, locals);
```

Comment:
The reason for this is to prevent the user having to do this themselves and to allow for the possibility of the following:

```
<video ng-class="{fadeInNicely: vm.loaded}" on-loaded-metadata="vm.loaded = true" src="vid.mp4">